core/src/main/scala/com/softwaremill/diffx/DiffFor.scala

```diff
@@ -1,14 +1,29 @@
 package com.softwaremill.diffx
 
-trait DiffFor[T] {
-  def apply(left: T, right: T): DiffResult
+trait DiffFor[T] { outer =>
+  protected val ignored: List[List[String]] = List.empty
+
+  def apply(left: T, right: T): DiffResult = apply(left, right, Nil)
```

Comment:
why `Nil` and not `ignored`?

Reply:
It will still be overrided by line 10:
```
    override def apply(left: T, right: T, toIgnore: List[FieldPath]): DiffResult =
      outer.apply(left, right, toIgnore ++ ignored)
```

Reply:
when do you use `ignored` then?

Reply:
Again line 10: 
```
    override def apply(left: T, right: T, toIgnore: List[FieldPath]): DiffResult =
      outer.apply(left, right, toIgnore ++ ignored)
    override val ignored: List[FieldPath] = outer.ignored ++ List(fields.toList)
```
I combine ignored from upstream (toIgnore) with ignored from current DiffFor(ignored) with ignored from wrapped DiffFor(outer)